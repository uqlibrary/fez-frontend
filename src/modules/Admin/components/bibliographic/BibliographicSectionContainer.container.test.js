import { BibliographicSectionContainer, mapStateToProps } from './BibliographicSectionContainer';
import Immutable from 'immutable';

function setup(testProps = {}, args = { isShallow: true }) {
    const props = {
        formValues: {
            toJS: jest.fn(() => ({})),
        },
        ...testProps,
    };

    return renderComponent(BibliographicSectionContainer, props, args);
}

describe('BibliographicSectionContainer component', () => {
    it('should render default view', () => {
        const render = setup();
        expect(render.getRenderOutput()).toMatchSnapshot();
    });

    it('should render disabled view', () => {
        const render = setup({ disabled: true });
        expect(render.getRenderOutput()).toMatchSnapshot();
    });

    it('should map state to props', () => {
        expect(mapStateToProps({}, {})).toEqual({
            formValues: Immutable.Map({}),
        });
    });
});
