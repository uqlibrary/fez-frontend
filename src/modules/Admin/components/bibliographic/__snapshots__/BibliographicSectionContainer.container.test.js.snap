// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BibliographicSectionContainer component should render default view 1`] = `
<Context.Provider
  value={
    {
      "formValues": {},
    }
  }
>
  <Memo(BibliographicSection) />
</Context.Provider>
`;

exports[`BibliographicSectionContainer component should render disabled view 1`] = `
<Context.Provider
  value={
    {
      "formValues": {},
    }
  }
>
  <Memo(BibliographicSection)
    disabled={true}
  />
</Context.Provider>
`;
